### PR TITLE
Change 1.x branch to pull from 2.6 snapshot and revert 'Migrate client transports to Apache HttpClient/Core5.x'

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
These changes enable the 1.x branch of opensearch-sdk-java to be compatible with the 2.x branch of OpenSearch

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle
Comment (dbwiddis) [inline build.gradle:79]: How are we making sure we keep this updated with the latest OpenSearch 2.x releases?  

When 2.6.0 goes out, this would need to be updated to 2.7.0-SNAPSHOT, but there is no mechanism to let us know this.  A breaking change can go in on 2.7 that we would never see.

We should be able to use Gradle's [`ComponentSelectionRules`](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ComponentSelectionRules.html) with a `2+` version, only accepting versions with `SNAPSHOT` in the name.  That should get us the highest 2.x SNAPSHOT version.

File: build.gradle
Comment (dbwiddis) [inline build.gradle:95]: For my own knowledge, is there a compatibility matrix for `opensearch-java` vs. `OpenSearch`?

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (dbwiddis) [inline src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: Eww, not only do we need to change the dependency with import, we need to change code as well?  Or does any 2.x version work in this test?   

Might consider creating a system property in the `build.gradle` and pulling it in from here.

File: build.gradle
Comment (ryanbogan) [reply build.gradle:95]: Not up to date but it's here: https://github.com/opensearch-project/opensearch-java/blob/main/COMPATIBILITY.md

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (ryanbogan) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: That's a good idea

File: build.gradle
Comment (ryanbogan) [reply build.gradle:79]: Those rules allow us to read from the snapshot locations?

Comment (ryanbogan) [reply build.gradle:79]: I could have a variable for `opensearchVersion` that's just `2.6.0` and another variable for `opensearchSnapshot` that concatenates the version with `-SNAPSHOT`.  Then we just have to change the version whenever a release goes out, similar to bwc tests

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (ryanbogan) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: Potential solution in the comment above: https://github.com/opensearch-project/opensearch-sdk-java/pull/420#discussion_r1097997733

File: build.gradle
Comment (dbwiddis) [reply build.gradle:79]: I just edited the version like this:
```groovy
    def opensearchVersion = "2.+"
```
And got this:
```
➜  opensearch-sdk-java git:(revert_http_client) ✗ gradle dependencies | grep opensearch-core
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
```
The "+" follows a prefix and I believe matches the newest version with that prefix, be it release or snapshot.  You can further filter it, but I think "newest" woroks here.

Comment (dbwiddis) [reply build.gradle:79]: For "1.+" I get:
```
➜  opensearch-sdk-java git:(revert_http_client) ✗ gradle dependencies | grep opensearch-core
|    +--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
|    |    \--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
```

Comment (ryanbogan) [reply build.gradle:79]: Ok so we can change `opensearchVersion` to `2.+`, but we would still need another variable for the tests right?

Comment (dbwiddis) [reply build.gradle:79]: From [the docs](https://docs.gradle.org/current/userguide/single_versions.html):
> A _prefix_ version range: e.g. 1.+, 1.3.+
> 
> Only versions exactly matching the portion before the + are included.
>
> The range + on it’s own will include any version.

That includes snapshots (assuming the snapshot repo is included).  Also,
> Versions are ordered based on the following rules:
>
> Each version is split into it’s constituent "parts":
> 
> The characters [. - _ +] are used to separate the different "parts" of a version.

and then

> The equivalent parts of 2 versions are compared using the following rules:
> 
> If both parts are numeric, the highest numeric value is higher: 1.1 < 1.2
> 
> If one part is numeric, it is considered higher than the non-numeric part: 1.a < 1.1
> 
> If both are non-numeric, the parts are compared alphabetically, in a case-sensitive manner: 1.A < 1.B < 1.a < 1.b
> 
> A version with an extra numeric part is considered higher than a version without (even when it’s zero): 1.1 < 1.1.0
> 
> A version with an extra non-numeric part is considered lower than a version without: 1.1.a < 1.1

So what's happening is it's breaking it into 2, 6, 0, SNAPSHOT which is higher than any 2, 5, anything.  When 2.6.0 is released it will be on the top until 2, 7, anything is available.

So just put "2+". :)

(We should probably switch "main" to just "+" ;)

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (dbwiddis) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: I'm going to guess the exact version doesn't matter and we can leave this as is.

**Review by dbwiddis — APPROVED** [review]

File: build.gradle
Comment (dbwiddis) [reply build.gradle]: I"m not sure the version in the tests matter (since 2.6 isn't even out).  Does the calling code check the version?

Comment (ryanbogan) [reply build.gradle]: Yeah it checks with the recent minimum compatibility check

Comment (ryanbogan) [reply build.gradle]: I just checked and it works with previous minor versions, since it's the minimum compatibility.  So we won't have to change this ever

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (dbwiddis) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: Actually there's an equals comparison with CURRENT, I think.

So this would have been a nice red flag that we needed to update our version, but why not just set these lines to `Version.CURRENT`?

Comment (ryanbogan) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:117]: It's minimum compatibility so it'll pass as long as the OpenSearch version > than 2.6

Comment (ryanbogan) [reply src/test/java/org/opensearch/sdk/TestExtensionsRunner.java]: I don't see any reason not to change it to `Version.CURRENT` though, so I did

**Comment by dbwiddis** [comment]
Interesting, your latest push didn't invalidate my review.... oh... 2.X branch :)

**Comment by ryanbogan** [comment]
Yep I noticed the same thing, different rules for this branch.  I'm still gonna wait for a second approval though since this isn't a backport

**Review by dbwiddis — APPROVED** [review]

File: src/main/java/org/opensearch/sdk/SDKClient.java
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/SDKClient.java:107]: Any reason to remove mapper here? We are registering modules above already

File: build.gradle
Comment (owaiskazi19) [inline build.gradle:95]: Let's create a variable for the client version too

Comment (ryanbogan) [reply build.gradle]: Done

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**
